nyaa/api_torznab.py

```diff
@@ -101,71 +112,113 @@ def torznab_caps():
     return render_torznab_caps()
 
 
-def torznab_search():
-    page_size = app.config['RESULTS_PER_PAGE']
+def get_categories_filter(cat):
+    categories = []
+    if cat:
+        for item in cat.split(','):
+            # Parse 1_1 format for compatibility
```

Comment:
"Honey, can you go to the walmart pick something up from the fruits shelf, will ya?"
Guy comes home with some beer, chips and magazines... "Sorry honey, I swear, there was no fruits shelf so I assumed you didn't mind what I picked."

It _is_ the principle of least surprise, normally you would toss the specified categories in a db query, a simple many-to-many table that maps between category numbers and releases. Such a query would yield no results. And thus it shouldn't in this case either.

We're translating from external categories to internal ones, but from the user perspective the behavior should remain the same.